li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -55,14 +59,22 @@
   private UUID _clientId;
 
   // Per cluster Consumers
-  private Map<ClusterDescriptor, LiKafkaConsumer<K, V>> _consumers;
+  private volatile List<Map.Entry<ClusterDescriptor, LiKafkaConsumer<K, V>>> _consumers;
```

Comment:
Map.Entry can be (ab)used as a tuple, but i'd use a private static class of our own